packages/app/src/vis-packs/core/matrix/MappedMatrixVis.tsx

```diff
@@ -10,41 +10,40 @@ import shallow from 'zustand/shallow';
 
 import type { DimensionMapping } from '../../../dimension-mapper/models';
 import { useMappedArray, useSlicedDimsAndMapping } from '../hooks';
+import { getSliceSelection } from '../utils';
 import MatrixToolbar from './MatrixToolbar';
 import { useMatrixConfig } from './config';
 import { getCellWidth, getFormatter } from './utils';
 
 interface Props {
   dataset: Dataset<ArrayShape, PrintableType>;
-  selection: string | undefined;
   value: ArrayValue<PrintableType>;
-  dims: number[];
   dimMapping: DimensionMapping;
   toolbarContainer: HTMLDivElement | undefined;
 }
```

Comment:
Looking at what is needed to add support for nD, I realized that there were a few props in `MappedMatrixVis` that were derived from other props:
- `selection` can be derived from `dimMapping` through `getSliceSelection`
- `dims` is simply `dataset.shape`

Therefore, I think removing these props simplifies the component and reduces the risk of misuse.

Reply:
Hmm true, true, I was going to argue against calling `getSliceSelection` in `MappedCompoundMatrixVis`, but you're making an excellent point. `getSliceSelection` is cheap, so why not :shrug: 

Do you want to do this in the other containers/mapped vis, then?